cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -370,9 +368,51 @@ func (p *BicepProvider) Destroy(
 		})
 }
 
-func (p *BicepProvider) getResourceGroupsFromDeployment(ctx context.Context) ([]string, error) {
+// latestDeployment finds the most recent deployment for the given environment in the provided scope.
```

Comment:
Based on the results of the other other comment, it may be good if this comment would formally describe if the deployment is completed, or ongoing.

Reply:
Would also be good to describe that the result is guaranteed to be non-nil when `err == nil`, or we can dereference and pass struct value instead of ptr.

Reply:
I'm going to change this name to `latestCompletedDeployment` (and update the comments) so that the semantics here are if this gives you a deployment back that deployment was in a terminal state (it might have failed, but at least you know it's not in flight).